src/tokens/ERC1155.vy

```diff
@@ -198,23 +198,14 @@ def safeTransferFrom(owner: address, to: address, id: uint256, amount: uint256,
     @param data The maximum 1024-byte additional data
             with no specified format.
     """
-    assert owner == msg.sender or self.isApprovedForAll[owner][msg.sender], "ERC1155: caller is not token owner or approved"
```

Comment:
can we have that check on the `external` function level (reason is composability later once the modules feature is available in Vyper):

```vy
assert owner == msg.sender or self.isApprovedForAll[owner][msg.sender], "ERC1155: caller is not token owner or approved"
self._safe_transfer_from(owner, to, id, amount, data)
```

similar to how I've done it here: https://github.com/pcaversaccio/snekmate/blob/main/src/tokens/ERC721.vy#L383